source/val/validation_state.cpp

```diff
@@ -158,6 +158,7 @@ ValidationState_t::ValidationState_t(const spv_const_context ctx,
       local_vars_(),
       struct_nesting_depth_(),
       grammar_(ctx),
+      has_memory_model_specified_(false),
```

Comment:
I think it would be better to reuse `memory_model_`, but initialize it to SpvMemoryModelMax.